[MINOR] Improve docs by adding ToC links to Monitoring

KevinLiLu opened the pull request:
My top 2 reasons for visiting the Kafka docs are to:
- View configurations
- View metrics 

This PR aims to improve the user experience for viewing metrics:
- Add href links to the `Monitoring` section of the Table of Contents so users do not need to scroll or Ctrl/Cmd-F to find specific metric details (Monitoring section has grown large as more component & metrics are added)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

KevinLiLu commented:
Pinging @guozhangwang @rajinisivaram for a review.

viktorsomogyi commented on docs/toc.html:
```diff
@@ -106,6 +106,15 @@
+                        <li><a href="#new_consumer_monitoring">New Consumer Monitoring</a></li>
+                        <li><a href="#connect_monitoring">Connect Monitoring</a></li>
+                        <li><a href="#kafka_streams_monitoring">Streams Monitoring</a></li>
+                        <li><a href="#others_monitoring">Others</a></li>
```
In ops.html the `basic_ops_audit` section is in Monitoring. Do you think we should include that as well?

KevinLiLu replied:
Thanks for the review! 

I noticed that section as well and I think that section could use some refactoring/rewording. The mentioned JIRA [KAFKA-260](https://issues.apache.org/jira/browse/KAFKA-260) is very old from 0.8.0. The `audit` section mentions features similar to a data loss auditing pipeline, but it looks like it was never implemented in the end ("Won't Fix"). I believe what came out of that JIRA issue eventually was message timestamps.

Regardless, that section implies that Kafka itself provides such auditing features, but these features must be built on top of Kafka (by leveraging timestamps for example).

I believe we should change the wording to give a recommendation like the `Others` section, and include it to the ToC. Alternatively, we could remove the section altogether.

omkreddy replied:
Since the audit feature is not part of Kafka, I think we can remove the audit section.

viktorsomogyi reviewed (commented):
Hi Kevin, if you don't mind I reviewed it as well and left one comment.

KevinLiLu committed:
MINOR: Improve docs by adding ToC links for Monitoring sections

KevinLiLu commented:
@lindong28 Can you give this documentation change a review? Thanks!

omkreddy commented on docs/toc.html:
```diff
@@ -101,6 +101,16 @@
+                        <li><a href="#selector_monitoring">Selector Monitoring</a></li>
+                        <li><a href="#common_node_monitoring">Common Node Monitoring</a></li>
+                        <li><a href="#producer_monitoring">Producer Monitoring</a></li>
+                        <li><a href="#new_consumer_monitoring">New Consumer Monitoring</a></li>
```
we need to remove "new" from href and from title

omkreddy reviewed (commented):
@KevinLiLu Thanks for the PR. Left couple of minor comments

KevinLiLu committed:
Remove audit section, and remove 'new' from consumer monitoring title

KevinLiLu commented:
Hi @omkreddy, thanks for the review. I have rebased and added a commit to address the comments.

If you have 2 minutes to spare (I promise it will only take 2 minutes 😄 ), could you also give [KIP-351](https://cwiki.apache.org/confluence/display/KAFKA/KIP-351%3A+Add+--under-min-isr+option+to+describe+topics+command) a comment and vote? Thanks~

omkreddy reviewed (approved):
@KevinLiLu Thanks for the PR. LGTM.
